File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        List<Integer> list = new ArrayList<>();
        
        int idx = 0;
        while (idx < progresses.length) {
            
            // 하루치 진도 진행
            for (int i = 0; i < progresses.length; i++) {
                progresses[i] += speeds[i];
            }
            
            int count = 0;
            for (int i = idx; i < progresses.length; i++) {
                if (progresses[i] >= 100) {
                    count += 1;
                    idx += 1;
                } else {
                    break;
                }
            }
            
            if (count > 0) list.add(count);
        }
        
        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    
    idx = 0
    for day in range(1, 100):
        
        # 모든 작업이 완료되면 종료
        if idx >= len(progresses):
            break
        
        count = 0
        for i in range(idx, len(progresses)):
            p, s = progresses[idx], speeds[idx]
            if p + s * day >= 100:
                count += 1
                idx += 1
            else:
                break
                
        if count > 0:
            answer.append(count)
        
    
    return answer